packages/react-email/build-preview-server.mjs

```diff
@@ -0,0 +1,24 @@
+import { spawn } from 'node:child_process';
+import { existsSync, promises as fs } from 'node:fs';
+
+const nextBuildProcess = spawn('next', ['build'], {
+  detached: true,
+  stdio: "inherit"
+});
+
+process.on('SIGINT', () => {
+  nextBuildProcess.kill('SIGINT');
+});
+
+nextBuildProcess.on('exit', async (code) => {
+  if (code !== 0) {
+    process.exit(code);
+  }
+
+  if (existsSync('dist/preview')) {
+    await fs.rm('dist/preview', { recursive: true });
```

Comment:
we could probably use the sync version of these functions and have the `exit` callback not being `async`